arrows/core/video_input_split.cxx

```diff
@@ -254,6 +254,45 @@ ::next_frame( kwiver::vital::timestamp& ts,   // returns timestamp
 } // video_input_split::next_frame
 
 
+// ------------------------------------------------------------------
+kwiver::vital::timestamp
+video_input_split
+::frame_timestamp() const
+{
+  // Check for at end of data
+  if ( this->end_of_video() )
+  {
+    return {};
+  }
+
+  auto const& image_ts = d->d_image_source->frame_timestamp();
+  auto const& metadata_ts = d->d_metadata_source->frame_timestamp();
+
+  auto ts = metadata_ts;
+  if ( image_ts != metadata_ts )
+  {
+    if ( image_ts.get_frame() == metadata_ts.get_frame() )
+    {
+      if ( image_ts.has_valid_time() && ! metadata_ts.has_valid_time() )
+      {
+        ts.set_time_usec( image_ts.get_time_usec() );
+      }
```

Comment:
It seems we should also handle the case where `metadata_ts` has valid time, but `image_ts` does not and then we take the time from `metadata_ts`.

Reply:
Sure, that seems plausible, but here I just copied what `next_frame` does. If we change one, we should change the other. That could be a separate PR. Would you prefer I make that change now?

Reply:
Oh I see.  Yes I think it makes sense to make that change on this branch, and given that there are 20 lines or so of non-trivial code in common, it probably also makes sense to factor our a helper function that is called by both `next_frame` and `frame_timestamp` to merge the timestamps.  That way these functions are less likely to get out of sync.

Reply:
Done.